src/main/scala/org/apache/spark/deploy/armada/Config.scala

```diff
@@ -86,4 +86,38 @@ private[spark] object Config {
       .stringConf
       .checkValue(selectorsValidator, "Selectors must be valid kubernetes labels/selectors")
       .createWithDefaultString(DEFAULT_CLUSTER_SELECTORS)
+
+  private[armada] val singleLabelOrNone: Regex = (s"^($label)?$$").r
+  private[armada] val singleLabelValidator: CharSequence => Boolean = l => {
+    val labelMaybe = singleLabelOrNone.findPrefixMatchOf(l)
+    labelMaybe match {
+      case Some(l) => true
+      case None => false
+    }
+  }
+
+  val GANG_SCHEDULING_NODE_UNIFORMITY_LABEL: ConfigEntry[String] =
+    ConfigBuilder("spark.armada.scheduling.nodeUniformityLabel")
+      .doc("A single kubernetes label to apply to both driver and executors")
+      .stringConf
+      .checkValue(singleLabelValidator, "Label must be a valid kubernetes label, just the label!")
+      .createWithDefaultString("armada-spark")
+
+  private val rfc1035labelPrefix: Regex = "([a-z][0-9a-z-]{0,29})?".r
```

Comment:
I don't love the rfc # in the val name.  Couldn't we call it something like validDomainNamePrefix?

Reply:
Addressed in bfc1a31